File: C/columns.c
#include <columns.h>

// Number of result columns
extern const uint8_t COLUMNS;

// Result output format
extern const char *RESULT_FORMAT;


void print_column(FILE *output, unsigned int result, uint32_t *result_counter_ref)
{
    uint32_t result_counter = *result_counter_ref;

    if (result_counter % COLUMNS != 0)
    {
        fputc(' ', output);
    }
    fprintf(output, RESULT_FORMAT, result);
    ++result_counter;
    if (result_counter % COLUMNS == 0)
    {
        fputc('\n', output);
    }

    *result_counter_ref = result_counter;
}


void finish_column(FILE *output, uint32_t *result_counter_ref)
{
    if (*result_counter_ref % COLUMNS != 0)
    {
        fputc('\n', output);
    }
}

File: Java/Columns.java
public class Columns
{
    private int resultCounter;

    // Number of result columns
    final private int COLUMNS;

    // Result output format
    final private String FORMAT;


    public Columns(int columns, String resultFormat)
    {
        resultCounter = 0;
        COLUMNS       = columns;
        FORMAT        = resultFormat;
    }


    public void printColumn(int result)
    {
        if (resultCounter % COLUMNS != 0)
        {
            System.out.print(' ');
        }
        ++resultCounter;
        System.out.printf(FORMAT, result);
        if (resultCounter % COLUMNS == 0)
        {
            System.out.println();
        }
    }


    public void finishColumn()
    {
        if (resultCounter % COLUMNS != 0)
        {
            System.out.println();
        }
    }


    public int getResultCount()
    {
        return resultCounter;
    }
}
